### ISSUE TITLE
Inaccessible "add block" with keyboard in editor (Accessibility [a11y])

### ISSUE THREAD

**Issue by iDschepe** [WordPress/gutenberg#51522]
### Description

It's not possible to access the "Add block" feature via keyboard.
Together with @Ryokuhi (Thanks for your help!) we tried to figure out what caused this issue at WCEU2023 contributer day. 

It seems to be an issue of `tabindex`

I guess the affected component might be within `PrivateInserter`: https://github.com/WordPress/gutenberg/blob/f1d0bd550f85f5fa4279a3fdb9a2b9c28a7544c6/packages/block-editor/src/components/inserter/index.js#L79

### Step-by-step reproduction instructions

1. Open gutenberg editor
2. Create a block, e.g. Social Icons block
3. Try to EDIT existing Social Icons block and add a new social icon item by accessing the inserter, using your keyboard

### Screenshots, screen recording, code snippet

![block-editor](https://github.com/WordPress/gutenberg/assets/40547587/47634ee6-db97-4736-97c2-a563c0e2021d)

### Environment info

See also:  https://wordpress.github.io/gutenberg/?path=/story/playground-block-editor--default to inspect `tabindex` of the inserter

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by iDschepe** [comment]
Ping

**Comment by henkwhite** [comment]
I am able to reproduce the bug. 

![Image](https://github.com/user-attachments/assets/7fdeb120-0095-4986-8c39-e2fcff0cf784)

**Comment by SirLouen** [comment]
## Reproduction Report
### Description
❌ This report can't validate that the issue can be reproduced.

### Environment
- WordPress: 6.9-alpha-60093-src
- PHP: 8.2.29
- Server: Apache/2.4.62 (Debian)
- Database: mysqli (Server: 11.8.2-MariaDB-ubu2404 / Client: mysqlnd 8.2.29)
- Browser: Chrome 138.0.0.0
- OS: Windows 10/11
- Theme: Twenty Twenty-Five 1.3
- MU Plugins: None activated
- Plugins:
  * Gutenberg 21.3.0-rc.2
  * Test Reports 1.2.0

### Testing Instructions
1. Go to posts
2. Try to add a new Social Icon with the keyboard
3. Try to edit the Social Icon with the keyboard
4. ❌ The Social Icon is editable with the keyboard
- Screencast to showcase provided in Supp Artifacts

### Actual Results
1.  ❌  Error condition is not occurring

### Additional Notes
- I can't really see where is the problem @henkwhite @iDschepe can you review my screencast and steps and see what I'm doing wrongly? 
- I'm going to tag this for now as `Needs Test Info` in case someone could give me for test data to verify if I can reproduce this issue
- By the way @henkwhite @iDschepe don't use GIF for reproduction, they look cool, but they are a hassle to analyse. I cannot fast-forward them or pause them. Just drop in the box the video screencast without converting it into GIF.

### Supplemental Artifacts
Screencast: https://f003.backblazeb2.com/file/wordpress-videos/wp-videos/2025/07/gb51522.mp4

**Comment by SirLouen** [comment]
Given that this has not been reproducible and more than 2 weeks have passed since it was tagged `Needs More Info` it's time to close this ticket.

@iDschepe please feel free to comment if you happen to find new reproduction information. I will be glad to review it and test it again, and in case it is reproducible this ticket will be reopened and an attempt to be fixed.
